.github/workflows/pre-post-release.yml

```
@@ -0,0 +1,97 @@
+---
+
+name: Pre/Post Release
+    
+on:
+  workflow_call:
+    inputs:
+      ref:
+        description: 'Branch or tag ref to run the workflow on'
+        type: string
+        required: true
+        default: "main"
+      version:
+        description: 'The version to release (e.g. 1.2.3). This workflow will automatically perform the required version bumps'
+        type: string
+        required: true
+      phase:
+        description: 'Pre or post release phase'
+        type: string # valid values are 'pre' or 'post'
+        required: true
+
+env:
+  RELEASE_VERSION: ${{ inputs.version }}
+  BRANCH_NAME: ${{ inputs.phase }}-release-v${{ inputs.version }}
+  
+permissions:
+  contents: read
+  
+jobs:
+  validate-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Validate release tag does not exist in repo
+        uses: ./.github/workflows/validate-tag
+        with:
+          tag: v${{ env.RELEASE_VERSION }}
```

If possible I'd suggest to use an env variable

```suggestion
          tag: ${{ env.RELEASE_VERSION_TAG }}
```

---

Ref: e151fba4c768e8fb520f4226102232e4a11e2426